Compile error, deprecated-copy, with GCC 9.1.0

### What version of gRPC and what language are you using?

1.22.0, cpp
 
### What operating system (Linux, Windows,...) and version?

Arch Linux; fully up to date as of 2019-07-05 15:21:30.
 
### What runtime / compiler are you using (e.g. python version or version of gcc)

```
$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-linux-gnu/9.1.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /build/gcc/src/gcc/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++ --enable-shared --enable-threads=posix --with-system-zlib --with-isl --enable-__cxa_atexit --disable-libunwind-exceptions --enable-clocale=gnu --disable-libstdcxx-pch --disable-libssp --enable-gnu-unique-object --enable-linker-build-id --enable-lto --enable-plugin --enable-install-libiberty --with-linker-hash-style=gnu --enable-gnu-indirect-function --enable-multilib --disable-werror --enable-checking=release --enable-default-pie --enable-default-ssp --enable-cet=auto
Thread model: posix
gcc version 9.1.0 (GCC)
```
 
### What did you do?

Compiling the project with GCC 9.1.0 fails due to the following error:

```
[CXX]     Compiling src/cpp/client/generic_stub.cc
In file included from include/grpcpp/impl/codegen/server_context_impl.h:28,
                 from include/grpcpp/impl/codegen/server_context.h:22,
                 from include/grpcpp/impl/codegen/async_stream.h:25,
                 from include/grpcpp/support/async_stream.h:22,
                 from include/grpcpp/generic/generic_stub_impl.h:24,
                 from include/grpcpp/generic/generic_stub.h:22,
                 from src/cpp/client/generic_stub.cc:21:
include/grpcpp/impl/codegen/call_op_set.h: In instantiation of ‘grpc::Status grpc::internal::CallOpSendM
essage::SendMessage(const M&, grpc::WriteOptions) [with M = grpc::ByteBuffer]’:
include/grpcpp/impl/codegen/call_op_set.h:407:45:   required from ‘grpc::Status grpc::internal::CallOpSendMessage::SendMessage(const M&) [with M = grpc::ByteBuffer]’
include/grpcpp/impl/codegen/async_unary_call.h:170:5:   required from ‘grpc::ClientAsyncResponseReader<R>::ClientAsyncResponseReader(grpc::internal::Call, grpc::ClientContext*, const W&, bool) [with W = grpc::ByteBuffer; R = grpc::ByteBuffer; grpc::ClientContext = grpc_impl::ClientContext]’
include/grpcpp/impl/codegen/async_unary_call.h:87:12:   required from ‘static grpc::ClientAsyncResponseReader<R>* grpc::internal::ClientAsyncResponseReaderFactory<R>::Create(grpc::ChannelInterface*, grpc::CompletionQueue*, const grpc::internal::RpcMethod&, grpc::ClientContext*, const W&, bool) [with W = grpc::ByteBuffer; R = grpc::ByteBuffer; grpc::CompletionQueue = grpc_impl::CompletionQueue; grpc::ClientContext = grpc_impl::ClientContext]’
src/cpp/client/generic_stub.cc:69:60:   required from here
include/grpcpp/impl/codegen/call_op_set.h:381:18: error: implicitly-declared ‘grpc::WriteOptions& grpc::WriteOptions::operator=(const grpc::WriteOptions&)’ is deprecated [-Werror=deprecated-copy]
  381 |   write_options_ = options;
      |   ~~~~~~~~~~~~~~~^~~~~~~~~
include/grpcpp/impl/codegen/call_op_set.h:88:3: note: because ‘grpc::WriteOptions’ has user-provided ‘grpc::WriteOptions::WriteOptions(const grpc::WriteOptions&)’
   88 |   WriteOptions(const WriteOptions& other)
      |   ^~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

--- a/include/grpcpp/impl/codegen/call_op_set.h
+++ b/include/grpcpp/impl/codegen/call_op_set.h
@@ -88,6 +88,9 @@ class WriteOptions {
   WriteOptions(const WriteOptions& other)
       : flags_(other.flags_), last_message_(other.last_message_) {}
 
+  /// Default assignment operator
+  WriteOptions& operator=(const WriteOptions& other) = default;
+
   /// Clear all flags.
   inline void Clear() { flags_ = 0; }
 

